test/e2e/set_delete_test.go

```diff
@@ -0,0 +1,252 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	appsv1 "k8s.io/api/apps/v1"
+	"strings"
+	"testing"
+	"time"
+
+	routev1 "github.com/openshift/api/route/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+var podCount int
+
+func testHeaders(t *testing.T, image string, route *routev1.Route, address string, headers []string, expectedResponse string, expectedMatches int) {
+	t.Helper()
+
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		t.Fatalf("failed to get kube config: %v", err)
+	}
+	client, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatalf("failed to create kube client: %v", err)
+	}
```

Comment:
I'm curious why you need to define a new client instead of using kclient.  It is only used when getting logs, and all other api calls you're using the global kclient.

Reply:
they both have different use cases.
kclient- 
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client@v0.15.0
client-
https://pkg.go.dev/k8s.io/client-go/kubernetes@v0.27.2

https://pkg.go.dev/k8s.io/client-go/kubernetes@v0.27.2 provide options to work with corev1 to use options like PodLogOptions which the client of controller-runtime does not have.